### PR TITLE
fix: ci, code refactoring, update DB struct, bugfixes

### PR DESCRIPTION
**Pull request by Rushaway**
Fix: #9 #10 #11

### CODE REVIEW

File: addons/sourcemod/scripting/KnockbackRestrict.sp [hunk 1]
```diff
@@ -339,18 +336,18 @@ void OnClientPostAdminCheck_Query(Database db, DBResultSet results, const char[]
...
+			if(strcmp(tempInfo.clientIP, "Unknown", false) == 0) {
 				/* Update IP to DB */
-				char query[120];
-				g_hDB.Format(query, sizeof(query), "UPDATE `KbRestrict_CurrentBans` SET `client_ip`='%s' WHERE `id`=%d", clientIP, tempInfo.id);
+				char query[MAX_QUERIE_LENGTH];
+				g_hDB.Format(query, sizeof(query), "UPDATE `KbRestrict_CurrentBans` SET `client_ip`='%s' WHERE `id`=%d", g_sIPs[client], tempInfo.id);
 				g_hDB.Query(OnUpdateClientIP, query);
 
-				FormatEx(tempInfo.clientIP, 20, "%s", clientIP);
+				FormatEx(tempInfo.clientIP, sizeof(Kban::clientIP), "%s", g_sIPs[client]);
```
Comment (maxime1907) [inline addons/sourcemod/scripting/KnockbackRestrict.sp:345]: its better to just put the src in sizeof like `sizeof(tempInfo.clientIP)`

File: addons/sourcemod/scripting/KnockbackRestrict.sp [hunk 2]
```diff
@@ -445,83 +437,75 @@ void OnGetKbansNumber(Database db, DBResultSet results, const char[] error, int
...
-		return;
-	}
-
-	if(g_bIsClientRestricted[client]) {
-		return;
-	}
+public Action Event_OnPlayerName(Handle event, const char[] name, bool dontBroadcast)
+{
+	int client = GetClientOfUserId(GetEventInt(event, "userid"));
+	if (client > 0 && IsClientInGame(client) && !IsFakeClient(client) && !IsClientSourceTV(client))
```
Comment (maxime1907) [inline addons/sourcemod/scripting/KnockbackRestrict.sp:451]: client <= MaxClients

File: addons/sourcemod/scripting/KnockbackRestrict.sp [hunk 3]
```diff
@@ -445,83 +437,75 @@ void OnGetKbansNumber(Database db, DBResultSet results, const char[] error, int
...
+		GetEventString(event, "newname", g_sName[client], sizeof(g_sName[]));
```
Comment (maxime1907) [inline addons/sourcemod/scripting/KnockbackRestrict.sp:452]: sizeof(g_sName[client])

File: addons/sourcemod/scripting/KnockbackRestrict.sp [hunk 4]
```diff
@@ -1634,58 +1508,31 @@ void OnKbanAdded(Database db, DBResultSet results, const char[] error, int array
...
-	}
 
-	if(cvar == null) {
-		// this should never happen
+	if(cvar == null) // Should never happen
 		return g_cvDefaultLength.IntValue;
-	}
 
-	if(time > cvar.IntValue) {
+	if(time == -1 || time > cvar.IntValue)
```
Comment (maxime1907) [inline addons/sourcemod/scripting/KnockbackRestrict.sp:1530]: time < 0

File: addons/sourcemod/scripting/KnockbackRestrict.sp (hunk 3 above)
Comment (Rushaway) [reply addons/sourcemod/scripting/KnockbackRestrict.sp:452]: No it will throw this error.
`error 001: expected token: "]", but found "-identifier-"`

Comment (maxime1907) [reply addons/sourcemod/scripting/KnockbackRestrict.sp:452]: okay

**Pull request merged**